### PR TITLE
feat(console): multi-text-input

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): multi-text-input
#### StringFiledArrayPath
This type is used to filter out the key of the `string[]` field. eg.
```ts
type Data = {
  a: string[];
  b: string;
  c: {
    d: string[];
  }
}

type ArrayPath = StringFiledArrayPath<Data>;   //  'a' | 'c.d'
```
image

## Linear Issue Reference

* LOG-1922

### Basic
multi-text-input

### Focus on the invalid input automatically
auto-focus

### CODE REVIEW

File: packages/console/src/components/MultiTextInput/MultiTextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/MultiTextInput.tsx:1]: MultiTextInput.tsx => index.tsx

File: packages/console/src/components/MultiTextInput/MultiTextInput.module.scss [hunk 2]
```diff
@@ -14,4 +14,10 @@
       margin-right: _.unit(3);
     }
   }
+
+  .error {
+    font: var(--font-body-2);
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/MultiTextInput.module.scss:19]: Please be noted that `--font-body-2` has been updated to `--font-body-medium` in the latest design. I have a PR to update the fonts in our code base (#462). If #462 gets merged before your PR, please remember to update the fonts used in your code

File: packages/phrases/src/locales/en.ts [hunk 3]
```diff
@@ -127,6 +127,8 @@ const translation = {
       delete: 'Delete',
       application_deleted: 'The application {{name}} deleted.',
       save_success: 'Saved!',
+      redirect_uri_required: 'You have to enter at least one redirect URI.',
+      no_space_in_uri: 'Sapce is not allowed in URI',
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/phrases/src/locales/en.ts]: fixed

File: packages/phrases/src/locales/zh-cn.ts [hunk 4]
```diff
@@ -127,6 +127,8 @@ const translation = {
       delete: 'Delete',
       application_deleted: 'The application {{name}} deleted.',
       save_success: 'Saved!',
+      redirect_uri_required: 'You have to enter at least one redirect URI.',
+      no_space_in_uri: 'Sapce is not allowed in URI',
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/phrases/src/locales/zh-cn.ts]: fixed

File: packages/console/src/components/MultiTextInput/MultiTextInput.module.scss (hunk 2 above)
Comment (demonzoo) [reply packages/console/src/components/MultiTextInput/MultiTextInput.module.scss]: Update: #462 has been merged. So please update this font usage.

Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/MultiTextInput.module.scss]: Will rebase & updated before merge

File: packages/console/src/components/MultiTextInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+  ? IsTuple<T> extends true
+    ? {
+        [K in TupleKeys<T>]-?: StringArrayPathImpl<K & string, T[K]>;
+      }[TupleKeys<T>]
+    : StringArrayPathImpl<ArrayKey, V>
+  : {
+      [K in keyof T]-?: StringArrayPathImpl<K & string, T[K]>;
+    }[keyof T];
+
+type StringFiledArrayPath<TFieldValues extends FieldValues> = StringArrayPath<TFieldValues>;
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/index.tsx:40]: Mind writing some comments explaining these types?

File: packages/console/src/components/MultiTextInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+};
+
+type MutiTextInputErrors = {
+  inputs: Record<number, string | undefined>;
+  required: string | undefined;
+};
+
+type Props<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends StringFiledArrayPath<TFieldValues> = StringFiledArrayPath<TFieldValues>
```
Comment (demonzoo) [inline packages/console/src/components/MultiTextInput/index.tsx:54]: `Filed` or `Field`?

File: packages/console/src/components/MultiTextInput/MultiTextInput.module.scss (hunk 2 above)
Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/MultiTextInput.module.scss]: updated

Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/MultiTextInput.module.scss]: updated

File: packages/console/src/components/MultiTextInput/index.tsx (hunk 6 above)
Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/index.tsx:63]: Align with the naming of react-hook-form.

File: packages/console/src/components/MultiTextInput/index.tsx (hunk 5 above)
Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/index.tsx]: added.

File: packages/console/src/components/MultiTextInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+> = {
+  control: Control<TFieldValues>;
+  name: TName;
+  isRequired?: boolean;
+  requiredMessage?: string;
+  rule?: InputsRule;
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/index.tsx]: updated

File: packages/console/src/components/MultiTextInput/index.tsx (hunk 6 above)
Comment (xiaoyijun) [reply packages/console/src/components/MultiTextInput/index.tsx]: updated

**Comment by xiaoyijun** [comment]
Closed for a better solution.
https://github.com/logto-io/logto/pull/472

**Pull request closed without merging**